galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -301,10 +303,26 @@ def move_content(self, request, *args, **kwargs):
         add_task = self._add_content(collection_version, dest_repo)
         remove_task = self._remove_content(collection_version, src_repo)
 
+        if settings.GALAXY_DEPLOYMENT_MODE == DeploymentMode.INSIGHTS.value:
+            golden_repo = AnsibleDistribution.objects.get(
+                base_path=settings.GALAXY_API_DEFAULT_DISTRIBUTION_BASE_PATH
+            ).repository
+
+            if dest_repo == golden_repo or src_repo == golden_repo:
+                repo_name = golden_repo.name
+                locks = [repo_name]
```

Comment:
The other `locks` in this module use the repository object rather than name... following this pulp_ansible pattern: https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/app/galaxy/mixins.py#L18

On a side note I see https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/tasks/synclist.py#L72 uses str of an object id... I had assumed `locks` to require an object, or if id was accepted it would need to be a pulp id since a pulp id is resolvable across models, but I'm not sure.

Reply:
I agree, it's odd. I'm not sure really which is correct. I've switched it back and fourth a few times. 

At one point the advice given to me was that locks only needed to be strings uniq to an object (ie, more or less a repr),
and that using objects may cause deadlocks.

But looking at more pulpcore code, a lot of it uses objects.  And the task results 'reserved resources' stuff certainly makes more sense when the locks are objects.

Reply:
Just updated it to use the Repository itself as the 'lock' / reserved resource. Seems to work fine.